package/MDAnalysis/analysis/base.py

```diff
@@ -170,12 +175,15 @@ def run(self, start=None, stop=None, step=None, verbose=None):
             number of frames to skip between each analysed frame
         verbose : bool, optional
             Turn on verbosity
+            
+        .. versionchanged:: 1.0.0
+            Verbose argument is now passed explicitly to
+            :meth:`AnalysisBase._setup_frames`.
         """
         logger.info("Choosing frames to analyze")
         # if verbose unchanged, use class default
         verbose = getattr(self, '_verbose', False) if verbose is None else verbose
```

Comment:
To my understanding, there is no need to use `getattr` here. `getattr` and `setattr` are used on dynamic operations, not on static ones.

Reply:
> Hi @Purva-Chaudhari ,
> 
> Just a small comment on to properly develop on a fork. I see you have made the changes in your forked `develop` branch. This is not completely wrong, but it is best to branch out first to a new branch that addresses only the PR/issue in question. And then, Pull Request that branch to the main repository base. If you continue developing new features in the fork base branch, as per habit, it may get really problematic if you have to deal with complex merge conflicts in future situations.

Sorry was sort of new to open source, will ensure this henceforth

Reply:
No problem. I understood that. You did a very nice first attempt. :wink: that's why I commented on it.

Reply:
Here `self._verbose` may not be defined, so the `getattr` is necessary. Another possibility would be to wrap the call in a `try`, but it would be more cluttered.

Reply:
> To my understanding, there is no need to use `getattr` here. `getattr` and `setattr` are used on dynamic operations, not on static ones.

should I make it verbose = False if none else verbose ?

Reply:
isn't `self._verbose` defined in the `__init__`? [link](https://github.com/MDAnalysis/mdanalysis/blob/452655df8cbfd19e4f4dd52e848c8d238481e349/package/MDAnalysis/analysis/base.py#L110)

Reply:
It is indeed. My bad.